File: YSY/12_6/코드 처리하기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string code) {
    string answer = "";
    int mode=0;
    for (int i=0;i<code.size();i++){
        if (mode==0){
            if (code[i]!='1'){
                if (i%2==0){
                    answer+=code[i];
                }
            }
            else if (code[i]=='1'){
                mode=1;
            }
        }
        else if (mode==1){
            if (code[i]!='1'){
                if (i%2==1){
                    answer+=code[i];
                }
            }
            else if (code[i]=='1'){
                mode=0;
            }
        }
    }
    if (answer==""){
        answer="EMPTY";
    }
    return answer;
}

File: JJB/Programmers/231126/코드_처리하기.py
def solution(codes):
    answer = []
    mode = False
    
    for idx, code in enumerate(codes):
        if code == "1":
            mode = not mode
        elif not mode and idx%2==0:
            answer.append(code)
        elif mode and idx%2==1:
            answer.append(code)
            
    if answer:
        answer = ''.join(answer)
    else:
        answer = "EMPTY"
    
    return answer